jumpscale/packages/tfgrid_solutions/chats/extend_kube.py

```diff
@@ -0,0 +1,108 @@
+
+    @chatflow_step(title="Adding nodes")
+    def add_nodes(self):
+        zos = j.sals.zos.get()
+        workload = zos.workloads.get(self.master_wid)
+        metadata = j.sals.reservation_chatflow.reservation_chatflow.decrypt_reservation_metadata(workload.info.metadata)
+        metadata = j.data.serializers.json.loads(metadata)
+        pool_id = workload.info.pool_id
+        old_wids = j.sals.marketplace.solutions.get_workloads_by_uuid(metadata.get("solution_uuid"))
+        old_nodes = [wid.info.node_id for wid in old_wids if wid.info.result.state == State.Ok]
+        nodes, pools = deployer.ask_multi_pool_distribution(self, self.nodes_count + len(old_nodes), self.node_query)
+        nodes_pools_zip = list(zip(nodes, pools))
+        selected_nodes = list(filter(lambda x: x[0].node_id not in old_nodes, nodes_pools_zip))
+        if len(selected_nodes) < self.nodes_count:
+            self.stop(
+                f"Failed to find resources to deploy {self.nodes_count}, available nodes are: {len(selected_nodes)}"
+            )
+        new_nodes = selected_nodes[: self.nodes_count]
+        network_view = deployer.get_network_view(workload.network_id)
+        master_ip = workload.ipaddress
+
+        self.reservations = []
+        for node, pool_id in new_nodes:
+            res = deployer.add_network_node(workload.network_id, node, pool_id)
+            if res:
+                for wid in res["ids"]:
+                    success = deployer.wait_workload(wid, breaking_node_id=node.node_id)
+                    if not success:
+                        raise StopChatFlow(f"Failed to add node {node.node_id} to network {wid}")
+            ip_address = network_view.get_free_ip(node)
+            if not ip_address:
+                raise StopChatFlow(f"No free IPs for network {network_name} on the specifed node" f" {node_id}")
+
+            self.md_show_update(f"Deploying worker on node {node}")
+            # Add public ip
+            public_id_wid = 0
+            if self.enable_public_ip:
+                public_id_wid = deployer.get_public_ip(
+                    pool_id, node.node_id, solution_uuid=metadata.get("solution_uuid")
+                )
+
+            self.reservations.append(
+                deployer.deploy_kubernetes_worker(
+                    pool_id,
+                    node.node_id,
+                    workload.network_id,
+                    workload.cluster_secret,
+                    workload.ssh_keys,
+                    ip_address,
+                    master_ip,
+                    size=self.cluster_size,
+                    identity_name=None,
+                    description="",
+                    public_ip_wid=public_id_wid,
+                    **metadata,
+                )
+            )
+
+        for resv in self.reservations:
+            success = deployer.wait_workload(resv, self)
+            if not success:
+                raise DeploymentFailed(
```

Comment:
what is the plan in case of failed deployments? 
this will only stop the chatflow without even checking the state for other workloads